### PR TITLE
Make Graphviz engine configurable

### PR DESCRIPTION
**Pull request by markuspf**
I recently found myself wanting to use a different GraphViz renderer for `Splash`, so I added this option. Might be useful for others.

### CODE REVIEW

File: doc/display.xml [hunk 1]
```diff
@@ -69,6 +69,17 @@
           option must be specified in <A>str</A> or in <A>opt.type</A>.
         </Item>
 
+        <Mark>engine</Mark>
+        <Item>
+          this option can be used to specify the GraphViz engine to use
+          to render a <C>dot</C>. The valid choices are <C>"dot"</C>,
+          <C>"neato"</C>,<C>"circo"</C>, <C>"twopi"</C>, <C>"fdp"</C>,
```
Comment (wilfwilson) [inline doc/display.xml]: There should be a space after the first comma.

File: tst/standard/display.tst [hunk 2]
```diff
@@ -102,6 +102,9 @@ gap> dot{[1 .. 50]};
 #gap> Splash("string", rec(viewer := "xpdf"));
 #Error, Digraphs: Splash: usage,
 #the option <type> must be "dot" or "latex",
+#gap> Splash("string", rec(type := "dot", engine := "dott");
```
Comment (wilfwilson) [inline tst/standard/display.tst]: You're missing a closing bracket from your "test"

File: doc/display.xml [hunk 3]
```diff
@@ -69,6 +69,17 @@
           option must be specified in <A>str</A> or in <A>opt.type</A>.
         </Item>
 
+        <Mark>engine</Mark>
+        <Item>
+          this option can be used to specify the GraphViz engine to use
+          to render a <C>dot</C>. The valid choices are <C>"dot"</C>,
```
Comment (wilfwilson) [inline doc/display.xml]: `<C>dot</C>` -> `<C>dot</C> file` or `<C>dot</C> document`.

File: tst/standard/display.tst [hunk 4]
```diff
@@ -102,6 +102,9 @@ gap> dot{[1 .. 50]};
...
+#Error, Digraphs: Splash: usage,
+#the option <engine> must be "dot", "neato","twopi", "circo", "fdp", "sfdp", or "patchwork"
```
Comment (wilfwilson) [inline tst/standard/display.tst]: There's a missing space between neato and twopi.

**Comment by markuspf** [comment]
I think I addressed your remarks.

Can you clarify what you mean by "some of the comments don't seem to be useful with the examples I've tried so far"?

**Comment by markuspf** [comment]
Ah, its supposed to mean "commands", excuse my slowness.

According to its documentation `fdp`, is for drawing "undirected graphs":

```
fdp draws undirected graphs using a ``spring'' model. It relies on a force‐directed approach in the spirit of Fruchterman and  Reingold  (cf.  Soft‐
       ware‐Practice & Experience 21(11), 1991, pp. 1129‐1164).
```

Of course we could make splash more complicated to catch misuse; that'd be quite a lot of work for very little gain, or we could include bits of the graphviz documentation in the documentation of splash...

**Comment by wilfwilson** [comment]
Oh yes, I meant 'commands'. I didn't spot the typo until just now either. I don't think we need to make anything more complicated.

**Pull request merged**